pkgs/development/compilers/microscheme/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchgit, pkgs }:
+
+stdenv.mkDerivation rec {
+  name = "microscheme-${version}";
+  version = "2015-02-04";
+
+  # externalize url/rev/sha256 to permit easier override
+  rev = "64dab5dff1a1fb54070bfc9a6d1f1be06cd9ad19";
+  sha256 = "0i3n7vwfbqsf95cif3qjq5hj6npnpwaiajdifnjlrn9jzki8lg40";
+  url = https://github.com/ryansuchocki/microscheme.git;
+
+  src = fetchgit {
+    inherit rev;
+    inherit sha256;
+    inherit url;
+  };
+
+  buildInputs = [ pkgs.vim ];
```

Comment:
> Looks strange, it's ok? :)

I initially created an `xxd` derivation to avoid this because it's really a deps on the `xxd` binary installed by vim.
The xxd derivation drew vim, went to the `src/xxd` folder, `make` the xxd...

@bjornfor suggested to avoid this and directly depend on vim because disk is cheap.